src/components/storage-plan-modal.js

```diff
@@ -0,0 +1,101 @@
+import PropTypes from 'prop-types';
+import ByteInput from './byte-input.js';
+import { Form, Input, Modal } from 'antd';
+import { useTranslation } from 'react-i18next';
+import React, { useEffect, useState } from 'react';
+import { baseStoragePlanShape } from '../ui/default-prop-types.js';
+
+const FormItem = Form.Item;
+
+function StoragePlanModal({ isVisible, storagePlan, storagePlanNamesInUse, onOk, onCancel }) {
+  const [form] = Form.useForm();
+  const { t } = useTranslation('storagePlanModal');
+  const [loading, setLoading] = useState(false);
+
+  const nameValidationRules = [
+    {
+      required: true,
+      message: t('nameRequired'),
+      whitespace: true
+    },
+    {
+      validator: (_rule, value) => {
+        return value && storagePlanNamesInUse.includes(value)
+          ? Promise.reject(new Error(t('storagePlanNameIsInUse')))
+          : Promise.resolve();
+      }
+    }
+  ];
+
+  const maxBytesValidationRules = [
+    {
+      min: 0,
+      type: 'number',
+      message: t('maxBytesZeroOrPositive')
+    }
+  ];
+
+  useEffect(() => {
+    if (isVisible) {
+      form.resetFields();
+    }
+  }, [isVisible, form]);
+
+  const handleOk = () => {
+    form.submit();
+  };
+
+  const handleOnFinish = async ({ _id, name, maxBytes }) => {
+    try {
+      setLoading(true);
+      await onOk({ _id, name, maxBytes });
+      setLoading(false);
+    } catch (error) {
+      setLoading(false);
+    }
+  };
+
+  const initialValues = {
+    _id: storagePlan?._id || '',
+    name: storagePlan?.name || t('newStoragePlan'),
+    maxBytes: storagePlan?.maxBytes || 100 * 1000 * 1000
```

Comment:
Here we'd need to check `storagePlan ? storagePlan.maxBytes : 100 * 1000 * 1000` because it seems that when you don't set the bytes (or set them to zero) and then edit the plan, you see "100 MB" and not your given value.

Reply:
Good catch, thanks. I changed it to `??` instead of `||` which should do the trick.